src/shared/FeatureIntroControlContext/index.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from "react";
+
+/**
+ * These items are expected to be passed to the context provider
+ */
+interface FeatureIntroControlContextConfig {
+  id?: string;
+}
+
+/**
+ * Required: dismiss, heading, content
+ */
+interface FeatureIntroControl extends FeatureIntroControlContextConfig {
+  heading: React.ReactNode | undefined;
```

Comment:
NBD `React.ReactNode` includes `undefined` so these can just be `React.ReactNode`
```suggestion
  heading: React.ReactNode;
```